### PR TITLE
craftos-pc: add basic tests

### PR DESCRIPTION
**Pull request by tomodachi94**
###### Description of changes

Adds basic package tests. This should speed up the review process for future updates.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by geri1701** [comment]
Result of `nixpkgs-review pr 232159` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by tomodachi94** [comment]
> Result of `nixpkgs-review pr 232159` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

@geri1701 looks like your review output didn't copy correctly.

**Comment by tomodachi94** [comment]
It wasn't building before due to `~` not being writeable (which makes Pulse mad, among other things); fixed by defining `$HOME`, `$XDG_DATA_DIR`, and `$XDG_CONFIG_DIR` as files from the store.

**Comment by geri1701** [comment]
Result of `nixpkgs-review pr 232159` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>craftos-pc</li>
  </ul>
</details>

**Review by geri1701 — APPROVED** [review]

File: pkgs/applications/emulators/craftos-pc/default.nix [hunk 1]
```diff
@@ -59,6 +60,13 @@ stdenv.mkDerivation rec {
     cp -R ${craftos2-rom}/* $out/share/craftos
   '';
 
+  passthru.tests = {
+    eval-hello-world = callPackage ./test-eval-hello-world { };
+    eval-periphemu = callPackage ./test-eval-periphemu { };
+  };
+
+  doCheck = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/emulators/craftos-pc/default.nix:68]: Are there tests in the package? doCheck does not influence passthru.tests

File: pkgs/applications/emulators/craftos-pc/test-eval-hello-world/default.nix [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+stdenv.mkDerivation {
+  name = "craftos-pc-test-eval-hello-world";
+  meta.timeout = 60;
+  buildCommand = ''
+    export HOME=$(pwd)
+    mkdir $HOME/.local $HOME/.config
+    export XDG_CONFIG_DIR=$HOME/.config
+    export XDG_DATA_DIR=$HOME/.local
+    ${craftos-pc}/bin/craftos --headless --script ${./init.lua} | grep "Hello Nixpkgs!" > /dev/null
```
Comment (SuperSandro2000) [inline pkgs/applications/emulators/craftos-pc/test-eval-hello-world/default.nix:13]:
```suggestion
    craftos --headless --script ${./init.lua} | grep "Hello Nixpkgs!" > /dev/null
```
please add to nativeBuildInputs

File: pkgs/applications/emulators/craftos-pc/test-eval-periphemu/default.nix [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+}:
+stdenv.mkDerivation {
+  name = "craftos-pc-test-eval-periphemu";
+  meta.timeout = 60;
+  buildCommand = ''
+    export HOME=$(pwd)
+    mkdir $HOME/.local $HOME/.config
+    export XDG_CONFIG_DIR=$HOME/.config
+    export XDG_DATA_DIR=$HOME/.local
+    if ${craftos-pc}/bin/craftos --headless --script ${./init.lua} | grep -q "FAIL"; then
```
Comment (SuperSandro2000) [inline pkgs/applications/emulators/craftos-pc/test-eval-periphemu/default.nix:12]: same here

File: pkgs/applications/emulators/craftos-pc/default.nix (hunk 1 above)
Comment (tomodachi94) [reply pkgs/applications/emulators/craftos-pc/default.nix]: Whoops, my bad.

File: pkgs/applications/emulators/craftos-pc/test-eval-periphemu/default.nix [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{ stdenv
+, craftos-pc
+, grep
+}:
+stdenv.mkDerivation {
+  name = "craftos-pc-test-eval-periphemu";
+  meta.timeout = 60;
+  nativeBuildInputs = [ craftos-pc grep ]
```
Comment (SuperSandro2000) [inline pkgs/applications/emulators/craftos-pc/test-eval-periphemu/default.nix:8]:
```suggestion
  nativeBuildInputs = [ craftos-pc grep ];
```

File: pkgs/applications/emulators/craftos-pc/test-eval-hello-world/default.nix [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{ stdenv
+, craftos-pc
+, grep
+}:
+
+stdenv.mkDerivation {
+  name = "craftos-pc-test-eval-hello-world";
+  meta.timeout = 60;
+  nativeBuildInputs = [ craftos-pc grep ]
```
Comment (SuperSandro2000) [inline pkgs/applications/emulators/craftos-pc/test-eval-hello-world/default.nix:9]:
```suggestion
  nativeBuildInputs = [ craftos-pc grep ];
```

File: pkgs/applications/emulators/craftos-pc/test-eval-periphemu/default.nix (hunk 4 above)
Comment (tomodachi94) [reply pkgs/applications/emulators/craftos-pc/test-eval-periphemu/default.nix]: Fixed in latest push.

File: pkgs/applications/emulators/craftos-pc/test-eval-hello-world/default.nix (hunk 5 above)
Comment (tomodachi94) [reply pkgs/applications/emulators/craftos-pc/test-eval-hello-world/default.nix]: Fixed in latest push.

File: pkgs/applications/emulators/craftos-pc/test-eval-hello-world/default.nix [hunk 6]
```diff
@@ -0,0 +1,18 @@
+{ stdenv
+, craftos-pc
+, grep
+}:
+
+stdenv.mkDerivation {
+  name = "craftos-pc-test-eval-hello-world";
+  meta.timeout = 60;
+  nativeBuildInputs = [ craftos-pc grep ];
```
[the remark this answers is not in the document]
Comment (tomodachi94) [reply pkgs/applications/emulators/craftos-pc/test-eval-hello-world/default.nix:9]: Ah, whoops. I'll fix this when there's a new release.

**Pull request merged**